core/src/Streamly/Internal/Data/MutByteArray/Type.hs

```diff
@@ -101,58 +102,46 @@ touch (MutByteArray contents) =
 -- XXX Some functions in this module are "IO" and others are "m", we need to
 -- make it consistent.
 
--- XXX We can provide another API for "unsafe" FFI calls passing an unlifted
--- pointer to the FFI call. For unsafe calls we do not need to pin the array.
--- We can pass an unlifted pointer to the FFI routine to avoid GC kicking in
--- before the pointer is wrapped.
---
--- From the GHC manual:
---
--- GHC, since version 8.4, guarantees that garbage collection will never occur
--- during an unsafe call, even in the bytecode interpreter, and further
--- guarantees that unsafe calls will be performed in the calling thread. Making
--- it safe to pass heap-allocated objects to unsafe functions.
+-- | NOTE: this is deprecated because it can lead to accidental problems if the
+-- user tries to use it to mutate the array because it does not return the new
+-- array after pinning.
```

Comment:
Aha, I was reading the changes from bottom-up hence the irrelevant comments later.